common/Perf/Azure.Test.Perf/PerfTest.cs

```diff
@@ -1,19 +1,87 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using Azure.Core;
+using Azure.Core.Pipeline;
 using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Azure.Test.Perf
 {
     public abstract class PerfTest<TOptions> : IPerfTest where TOptions : PerfOptions
     {
+        private readonly HttpPipelineTransport _insecureTransport;
+
+        private readonly HttpClient _recordPlaybackHttpClient;
+        private readonly TestProxyPolicy _testProxyPolicy;
+
+        private string _recordingId;
+
         protected TOptions Options { get; private set; }
 
         public PerfTest(TOptions options)
         {
             Options = options;
+
+            if (Options.Insecure)
+            {
+                var transport = (new PerfClientOptions()).Transport;
+
+                // Disable SSL validation
+                if (transport is HttpClientTransport)
+                {
+                    _insecureTransport = new HttpClientTransport(new HttpClient(new HttpClientHandler()
+                    {
+                        ServerCertificateCustomValidationCallback = (message, cert, chain, errors) => true
+                    }));
+                }
+                else
+                {
+                    // Assume _transport is HttpWebRequestTransport (currently internal class)
+                    ServicePointManager.ServerCertificateValidationCallback = (message, cert, chain, errors) => true;
+
+                    _insecureTransport = transport;
+                }
+            }
+
+            if (Options.TestProxy != null)
+            {
+                if (Options.Insecure)
+                {
+                    _recordPlaybackHttpClient = new HttpClient(new HttpClientHandler()
+                    {
+                        ServerCertificateCustomValidationCallback = (message, cert, chain, errors) => true
+                    });
+                }
+                else
+                {
+                    _recordPlaybackHttpClient = new HttpClient();
+                }
+
+                _testProxyPolicy = new TestProxyPolicy(Options.TestProxy);
+            }
+        }
+
+        protected TClientOptions ConfigureClientOptions<TClientOptions>(TClientOptions clientOptions) where TClientOptions : ClientOptions
```

Comment:
Should we name the method something more specific like `ConfigureClientOptionsForTransportAndProxy()` or `ConfigureTransportAndProxyInClientOptions()` or something like that.

Reply:
I intentionally kept it very generic, since the framework might configure more options in the future, and I didn't want to need to rename the method (and update all the callers).